src/plugins/d3/renderer/hooks/useShapes/bar-x.tsx

```diff
@@ -96,60 +97,121 @@ export function prepareBarXSeries(args: Args) {
         svgContainer,
     } = args;
 
-    const stackedSeriesMap = group(series, (item) => item.stackId);
+    const ref = React.useRef<SVGGElement>(null);
 
-    const seriesData = series.map(({data}) => data).flat(2);
-    const minPointDistance = minDiff(seriesData.map((item) => Number(item.x)));
-
-    const result: React.ReactElement[] = [];
+    React.useEffect(() => {
+        if (!ref.current) {
+            return;
+        }
 
-    Array.from(stackedSeriesMap).forEach(([stackId, stackedSeries]) => {
-        const stackHeights: Record<string, number> = {};
-        stackedSeries.forEach((item, seriesIndex) => {
-            item.data.forEach((point, i) => {
-                const rectProps = getRectProperties({
-                    point,
-                    xAxis,
-                    xScale,
-                    yAxis,
-                    yScale,
-                    minPointDistance,
+        const svgElement = select(ref.current);
+        svgElement.selectAll('*').remove();
+
+        const xValues =
+            xAxis.type === 'category'
+                ? []
+                : series.reduce<number[]>((acc, {data}) => {
```

Comment:
I think we should use common methods for such purposes. We already have [one](https://github.com/gravity-ui/chartkit/blob/main/src/plugins/d3/renderer/utils/index.ts#L53C14-L53C36)